mlx-macros/src/lib.rs

```diff
@@ -142,3 +145,63 @@ pub fn generate_test_cases(input: TokenStream) -> TokenStream {
 
     TokenStream::from(tests)
 }
+
+/// Derive the `ModuleParameters` trait for a struct. Mark a field with `#[param]` attribute to
```

Comment:
Do you think we'll eventually want to have a `derive(Module)`?

Reply:
I don't think so because the `Module` implementation would include the `forward` method. The associated error type was an attempt to allow user to have more customized errors but is currently limited because the c ffi can only take an mlx `Exception`. One thing I think that we could change to the `Module` trait is to provide an default noop `training_mode` implementation, but I didn't do this because I felt like user might forget if this is not mandatory.